src/components/modules/TagPostsSingleColumn/PostList.js

```diff
@@ -0,0 +1,21 @@
+import React, { PureComponent } from "react";
+import { View } from "react-native";
+import { Post } from "components/modules/";
+
+type Props = {
+  posts: any,
+};
+
+class PostList extends PureComponent<Props> {
```

Comment:
I wonder if it will ever be confusing to have a bunch of components with the same name. Did you consider creating this in components/modules? It seems like it could just passthrough all the props besides posts, and maybe take marginTop prop as well.

Reply:
It's a question I also feel a certain degree of dissonance about.

`PostLists` are only ever called from sibling components within the folder where they are written, and I think it's fair to think of them like private methods within the module folder's namespace. I'd promote them to a shared component abstraction except they're pretty idiosyncratic to the component.

It's a little annoying to have two files with the same name open in an IDE, but VSCode at least tags the tab with the distinguishing folder name.

There's a Sandi Metz quote (I think from [this dope AF talk](https://www.youtube.com/watch?v=8bZh5LMaSmE)) that related to this question, and I which find alleviating to the turmoil, where she says something approximate to: "If you have a lot of things named with the same prefix or suffix then it's a good indication that there's a hidden class waiting for it's own definition"

I'm open to other points of view though

Appreciate the review dude!

Reply:
Well the way I see some Post props are already being promoted up to be PostList props, ie isDesktop. Why not just promote all of them? 

Actually it looks like the only other Post attribute that needs to be promoted is layoutVariant. And the only other difference I can see among the postlists is the marginTop. So maybe those could just be parameters?

MostRecent would use the generic postlist like:
```
renderMobile:
            <PostList posts={secondaryPosts} isDesktop={false} marginTop={20} layoutVariant="reduced" />
renderDesktop:
            <PostList posts={secondaryPosts} isDesktop={true} marginTop={20} layoutVariant="reduced" />￼
```

TagPostsSingleColumn:

```<PostList posts={secondaryPosts} marginTop={25} layoutVariant="medium" />￼```

Trending:
```
          <PostList posts={posts} isDesktop={false} marginTop={20} layoutVariant="reduced" />
```

Where do the marginTop numbers come from? How closely linked are they to the layoutVariant? If they are very closely linked than perhaps generic PostList component could automatically derive marginTop from the layoutVariant.

Wouldn't that quote increase the anxiety about having a lot of things named PostList? It makes me want to try to discover the hidden class that underlies the repeated structure.

Reply:
Dude! Hella appreciate the level of consideration you're giving this.

In general, the idea behind those types of segmentations is separation of concerns / readability. Hard to know upfront where the most appropriate seams between interfaces will end up being.

Not sure if this is part of your question, but the initial thinking in having discrete `renderMobile` and `renderDesktop` was to make it easier to revisit with responsive distinctions being more or less first-class, and also to avoid the spaghetti of excessive responsive conditionals crammed into the main render function.

I think it's easier to think about `PostLists` when they're broken out from the rest of the module, into more "single responsibility" components, albeit there is a certain degree of discomforting prop inheritance.

>Wouldn't that quote increase the anxiety about having a lot of things named PostList? It makes me want to try to discover the hidden class that underlies the repeated structure.

Hah! Golly -- so I guess I misunderstood your meaning -- I brought the quote up as an argument against the proposition of something like `RecentPostsPostList`, `TagPostPostList`, etc., but thinking about it now I agree it doesn't actually apply beyond "there are prefixes/suffixes" and is definitely a "by your logic" in favor of promoting a `PostList` abstraction -- sheesh -- chalk it up to my having gotten no sleep 🙃 

I think there's a decent argument that `PostList`s could be a shared component, perhaps branching off of `layoutVariant`?... but I don't know if it's worth it. Like -- for example -- the `RecentPost` module's desktop `PostList` is highly idiosyncratic (one of the larger things left to do) since it has to contend with variable text length vertical spacing issues in reference to the main post's image.

Mobile has pretty consistent `PostList`s, and in fact it's essentially the same even regardless of `layoutVariant`. I could see having something shared for that, but I feel like it'd create confusion between the respective desktop equivalents? Maybe not though -- could have a mobile `PostList`, and `PostGrid` for most of the Desktop cases. `RecentPosts` would need to have a special desktop `PostList`, but maybe that's okay. 

My inclination would be to hold off on the abstraction, keeping them local to the module folder, colocated with the desktop counterparts, and have the shared attributes (spacing) represented in constants values. I'm a little risk averse towards potentially premature/entangling abstractions, since there was a pretty hefty detour as a result of my doing that at the beginning with Posts.

>Where do the marginTop numbers come from? How closely linked are they to the layoutVariant? 

The `marginTop` used to have responsive logic which I chopped for now to expedite the Post refactor. My plans was to revisit those handful of little spacing issues in the next couple days. I will definitely think about the efficiencies around `PostList`.

I know that's a lot of text, which I hope doesn't douse interest in responding -- appreciate the thoughtful perspective/suggestions

PS Also regrets that this PR had so much flow bullshit -- meant to have that separate to avoid the clutter. Will definitely avoid in future.